public/app/features/variables/state/selectors.ts

```diff
@@ -12,7 +12,7 @@ export const getVariable = <T extends VariableModel = VariableModel>(
     if (throwWhenMissing) {
       throw new Error(`Couldn't find variable with id:${id}`);
     }
-    return undefined;
+    return (undefined as unknown) as T;
```

Comment:
this is beautiful :)

Reply:
Also, a strict null error unfortunately, the correct way would be to define a better return type. Should be `T | undefined` but that would cause a ripple effect. When we remove the old variable system, a lot of these strict null errors will become a lot clearer.